lib/sass/script/parser.rb

```diff
@@ -483,11 +483,21 @@ def arglist(subexpr, description)
             args << e if e
           end
 
-          return args, keywords, splat unless try_tok(:comma)
+          return args, keywords, splat unless no_trailing_comma?(description)
           e = assert_expr(subexpr, description)
         end
       end
 
+      def no_trailing_comma?(description)
```

Comment:
A cleaner solution than modifying the comma-detection would be to refactor the loop itself so that it's more like

```
while there's a subexpression:
  parse the subexpression
  return if we don't parse a comma
```